config/dynamic.c

```diff
@@ -526,6 +539,17 @@ run_dynamic_config(const char *server_path, struct dynamic_config *dy_conf)
 		goto free_sock;
 	}
 
+	/*
+	 * TODO Add support for the dynamic configuration running as Grantor.
+	 */
+	if (gk_conf != NULL) {
+		gk_conf_hold(gk_conf);
+		dy_conf->gk = gk_conf;
+	} else {
+		ret = -1;
+		goto free_sock;
+	}
```

Comment:
Drop the `else` clause since no error happened. It's fine to have `gk_conf == NULL`.

Only `gk_conf_hold(gk_conf);` should be under the `if` statement, `dy_conf->gk = gk_conf;` should be after the `if` statement.